checking whether place. is in the csv columns

newtewt opened the pull request:
Now there is a check to see if `place.` is in the list of columns to create a new place. If not place is treated as a string and passed along in the request which the weapp api will fetch the existing location instead of creating a new one. 

https://github.com/medic/medic-conf/issues/135

garethbowen commented on src/fn/create-users.js:
```diff
@@ -18,9 +18,8 @@ module.exports = (projectDir, couchUrl) => {
         const password = row[cols.indexOf('password')];
         const type     = row[cols.indexOf('type')];
-
+        const place    = cols.find(e => e.includes("place.")) ? prefixedProperties(cols, row, 'place.'  ) : row[cols.findIndex(x => x === 'place')];
```
This line is a bit complicated for me, and really long! I'm not a big fan of single letter variables although this is a personal preference and up for debate. Finally, I think it'd be easier to read if it was a simple `if/else` rather than a ternary.

In addition, running `cols.find` for each row is inefficient. The other code in this section should also be optimised (eg: `cols.indexOf('type')`). I don't know how much difference this will make in the real world, but it would probably be noticeable if uploading thousands of users.

So how about replacing it with something like this...

```js
...
const { cols, rows } = fs.readCsv(csvPath);
const usernameIndex = cols.indexOf('username');
const passwordIndex = cols.indexOf('password');
const typeIndex = cols.indexOf('type');
const placeIdIndex = cols.indexOf('place'); // may be not found
const contactPropertiesFn = prefixedProperties(cols, 'contact.');
const placePropertiesFn = prefixedProperties(cols, 'place.');
return rows.reduce((promiseChain, row) => {
  const username = row[usernameIndex];
  const password = row[passwordIndex];
  const type = row[typeIndex];
  const contact = contactPropertiesFn(row);
  const place = placeIdIndex ? row[placeIdIndex] : placePropertiesFn(row);
  
  ...
});

// NB this is updated too!
const prefixedProperties = (cols, prefix) => {
  const indices = {};
  cols.forEach(col => {
    if (col.startsWith(prefix)) {
      indices[col.substring(prefix.length)] = cols.indexOf(col);
    }
  });
  return (row) => {
    const result = {};
    Object.keys(indices).forEach(key => result[key] = row[indices[key]]);
    return result;
  };
};
```

I haven't tested any of this so it's probably buggy, and it could be further improved but hopefully it shows what I mean.

garethbowen commented on test/fn/create-users.spec.js:
```diff
@@ -1,66 +1,81 @@
-const api = require('../api-stub');
-const assert = require('chai').assert;
-const createUsers = require('../../src/fn/create-users');
+const api = require("../api-stub");
```
This file has been converted to use double quotes everywhere. Our style is single quotes. Can you switch it back, and maybe enable eslint so it'll fail in future?

newtewt commented:
@garethbowen  I have updated based on your comments. I added the eslintrc file from webapp as initial starting point and updated the files I changed based on it. I'm going to open a separate pull request to remove jshint and put in eslint as the test on the project. My first run seems like some updates to other files and eslint tuning might need to occur.
